@here/harp-mapview/lib/MapView.ts

```diff
@@ -2418,11 +2418,12 @@ export class MapView extends THREE.EventDispatcher {
         elevationRangeSource: ElevationRangeSource,
         elevationProvider: ElevationProvider
     ) {
-        // Try to remove incase this method was already called, will do nothing if it doesn't exist.
-        this.removeDataSource(elevationSource);
-        this.addDataSource(elevationSource);
+        // Add as datasource if it was not added before
+        const isPresent = this.m_tileDataSources.some((ds) => ds === elevationSource);
+        if (!isPresent) {
+            this.addDataSource(elevationSource);
+        }
         this.m_elevationRangeSource = elevationRangeSource;
-        this.m_elevationRangeSource.connect();
```

Comment:
This can be removed, because `dataSource.connect()` was already called by `addDataSource()`. Calling it twice creates some errors in the console.